src/Orleans/project.json

```diff
@@ -1,6 +1,7 @@
-{
+﻿{
   "dependencies": {
-    "Newtonsoft.Json": "7.0.1"
+    "Newtonsoft.Json": "7.0.1",
+    "System.Collections.Immutable": "1.1.36"
```

Comment:
I think that we could take this dependency for now. We already have some talks of how we might refactor dependencies in the future, so this might be a non-issue then, but for now let's add it, as it is clearly a valuable contribution.
Make sure to add this dependency in `Microsoft.Orleans.Core.nuspec` too